game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/pbem/SendTestEmailAction.java

```diff
@@ -0,0 +1,68 @@
+package games.strategy.engine.framework.startup.ui.posted.game.pbem;
+
+import com.google.common.base.Ascii;
+import games.strategy.engine.ClientFileSystemHelper;
+import games.strategy.engine.posted.game.pbem.IEmailSender;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Level;
+import javax.swing.JOptionPane;
+import javax.swing.SwingUtilities;
+import lombok.extern.java.Log;
+import org.triplea.swing.ProgressWindow;
+
+@Log
+public class SendTestEmailAction {
+
+  /** Tests the email sender. This must be called from the swing event thread */
+  void send(final String to) {
+    SwingUtilities.invokeLater(() -> sendWithProgressWindow(to));
+  }
+
+  private void sendWithProgressWindow(final String to) {
+    final ProgressWindow progressWindow =
+        new ProgressWindow(JOptionPane.getFrameForComponent(null), "Sending test email...");
```

Comment:
All remaining code, at this point and below is simply extracted with exception of the 'to' field being parameterized.